docs/rfcs/20210223_external_connectivity_in_k8s.md

```diff
@@ -0,0 +1,189 @@
+## How do we teach this?
+
+The Kafka API requires that each broker will be reachable from
+the client side. To expose Redpanda nodes, but not add additional
+latency to the communication each Redpanda process will have assigned
+unique host port. In case of cloud providers the host VM need to be
+publicly reachable and have opened some range ports. The operator
+would be responsible to orchestrate Kubernetes to achieve that.
+
+## Dictionary
+
+- Kubernetes Service type NodePort with empty selector that gives
+  unused port that is available across all k8s worker nodes
+- ServiceAccount with RBAC rule assigned to init container will
+  allow retrieving the node information from Kubernetes API server
+
+# Reference-level explanation
+
+( This is the technical portion of the RFC )
+
+## Interaction with other features
+
+- The kafka multi listener needs to be set up in each Redpanda
+  node, so that internal connectivity can stay as is. For external
+  connectivity the dedicated port would be opened.
+- The statefulset must bootstrap Redpanda cluster with or without
+  external connectivity.
+- The user intention defined in cluster.redpanda.vectorized.io custom
+  resource should be reconciled into set of ready to use address with
+  ports that can be reach from the internet. The status field will
+  be the place to look for the addresses.
+
+## Telemetry & Observability
+
+- metrics that can be anchor in Redpanda or in VM network stack
+    - Number of active connection
+    - Number of dropped connection
+    - Number of bytes processed
+- Tail latency observed by the clients
+- Additional probe that will live close to operator, but will try
+  reach the Redpanda Nodes by their external address
+
+## Corner cases dissected by example.
+
+None.
+
+## Detailed design - What needs to change to get there
+
+The host port need to be added in kafka port in the Statefulset
+definition. Service type node port needs to be added. The init
+container needs to know how to get the public IP of the node.
+New ServiceAccount need to be added with the RBAC rule for getting
+node information.
+
+## Detailed design - How it works
+
+Each Kubernetes node needs to be exposed to the internet in order
+to expose Redpanda node. Operator's responsibility is creation of
+Redpanda configuration that has correctly registered advertised
+Kafka API and assign correct unique and unused node port. With the
+help of Kubernetes Service type node port the unused port can be
+injected into container host port. This service doesn't have selector,
+so that no rule should be created in `iptable` or any other CNI
+implementation. Init container will retrieve from the Kubernetes API
+server information about node and its public IP. That container will
+then change the Redpanda configuration, so that internal and external
+ports and addresses will be in Kafka API and Advertised Kafka API
+section.
+
+## Drawbacks
+
+The nodes must be exposed to some other networks. This scope goes
```

Comment:
let's also say here the drawback around tls certs

Reply:
No, I put this in the [Unresolved questions
](https://github.com/vectorizedio/redpanda/pull/749/files#diff-0ea54f9ef8f1891694e57459769c954e7822a8748f9284f9054bf54120283dadR186).

The first proposal that was tighten with external-dns could resolve the issue. Now we need to land this feature and resolve both of them in separated RFC.

Reply:
there's just so many sections in the RFC that sounds the same to me. Thanks